### PR TITLE
Add specific return type for non existent node

### PR DESCRIPTION
**Pull request by ivanpauno**
Addresses https://github.com/ros2/ros2cli/pull/322#issuecomment-525364023

### CODE REVIEW

File: rmw/include/rmw/ret_types.h [hunk 1]
```diff
@@ -36,6 +36,10 @@ typedef int32_t rmw_ret_t;
 /// Incorrect rmw implementation.
 #define RMW_RET_INCORRECT_RMW_IMPLEMENTATION 12
 
+// rmw node specific ret codes in 2XX
+/// Failed to find node name
+#define RMW_RET_NON_EXISTENT_NODE_NAME 203
```
Comment (dirk-thomas) [inline rmw/include/rmw/ret_types.h]: Why 203 instead of maybe 20?

Comment (dirk-thomas) [reply rmw/include/rmw/ret_types.h]: I found the matching define in `rcl` so this number makes sense. Maybe mention `rcl` in the existing comment.

Comment (jacobperron) [reply rmw/include/rmw/ret_types.h]: Does it matter if they're matching? There's a [convert function](https://github.com/ros2/rcl/blob/bad86c204db17f2538a537366edf74b4d79645bd/rcl/src/rcl/common.c#L68-L69) anyway.

Seems arbitrary to me.

Comment (jacobperron) [reply rmw/include/rmw/ret_types.h]: Also, I think the name needs to be updated according to https://github.com/ros2/rcl/pull/492#discussion_r319266399

```diff
- RMW_RET_NON_EXISTENT_NODE_NAME
+ RMW_RET_NODE_NAME_NON_EXISTENT
```

Comment (ivanpauno) [reply rmw/include/rmw/ret_types.h]: > Does it matter if they're matching? There's a convert function anyway.
>
> Seems arbitrary to me.

It's arbitrary yes, but I think it have some sense.

> Also, I think the name needs to be updated according to ros2/rcl#492 (comment)
> ```
> - RMW_RET_NON_EXISTENT_NODE_NAME
> + RMW_RET_NODE_NAME_NON_EXISTENT
> ```

Thanks, forgot to push that commit.

File: rmw/include/rmw/ret_types.h [hunk 2]
```diff
@@ -36,6 +36,11 @@ typedef int32_t rmw_ret_t;
 /// Incorrect rmw implementation.
 #define RMW_RET_INCORRECT_RMW_IMPLEMENTATION 12
 
+// rmw node specific ret codes in 2XX
+/// Failed to find node name
+// Using same return code than in rcl
```
Comment (hidmic) [inline rmw/include/rmw/ret_types.h]: @ivanpauno why?

Comment (ivanpauno) [reply rmw/include/rmw/ret_types.h]: I prefer using the same code when possible, I don't think it's important

**Pull request merged**